docs/master/performance/query-caching.md

```diff
@@ -2,13 +2,51 @@
 
 In order to speed up GraphQL query parsing, the parsed queries can be stored in the Laravel cache.
 
+## Configuration
+
 Query caching is enabled by default.
-You can define cache store and cache duration, see `config/lighthouse.php`.
+You can disable it by setting `query_cache.enable` to `false` in `config/lighthouse.php`.
+
+You can control how query caching works through the option `query_cache.mode` in `config/lighthouse.php`.
+
+### Mode `store`
+
+Use an external shared cache through a Laravel cache store like Redis or Memcached.
+This is only recommended if your application can not write to the local filesystem.
+
+### Mode `opcache`
+
+Store parsed queries in PHP files on the local filesystem to leverage OPcache.
+This is recommended if your application is running on a single server instance with write access to a persistent local filesystem.
+
+### Mode `hybrid`
+
+Leverage OPcache, but use a shared cache store when local files are not found.
+This is recommended if your application is running on multiple server instances with write access to a persistent local filesystem.
+
+## Cache invalidation
+
+When using an external cache store, you can configure a TTL for cached queries.
+That way, old queries that are potentially unused will be removed from the cache automatically.
+
+When using the file based cache, you have to manually remove old cached queries.
+Run the following command periodically to remove old files:
+
+```shell
+php artisan lighthouse:clear-query-cache --hours=<TTL in hours>
+```
+
+Make sure you flush the query cache when you deploy an upgraded version of the `webonyx/graphql-php` dependency.
+When using an external cache, remove all keys for the configured store:
+
+```shell
+php artisan cache:clear <store name>
+```
 
-Make sure you flush the query cache when you deploy an upgraded version of the `webonyx/graphql-php` dependency:
+When using the file based cache, remove all cached query files:
 
 ```shell
-php artisan cache:clear
+php artisan lighthouse:clear-query-cache
 ```
```

Comment:
This section needs to be adapted to the new `mode` option, if we decide to go with that. As mentioned in https://github.com/nuwave/lighthouse/pull/2713#discussion_r2329434868, perhaps we can offload the complexity to the command.